src/server/ranch/BreedingMarket.cpp

```diff
@@ -0,0 +1,610 @@
+//
+// Created for Alicia Server
+//
+
+#include "server/ranch/BreedingMarket.hpp"
+#include "server/ServerInstance.hpp"
+
+#include <libserver/data/DataDirector.hpp>
+
+#include <spdlog/spdlog.h>
+
+#include <algorithm>
+
+namespace server
+{
+
+namespace
+{
+
+// Breeding grade restrictions
+// TODO: Make these configurable via server config
+constexpr uint8_t MinBreedingGrade = 4;
+constexpr uint8_t MaxBreedingGrade = 8;
+
+} // namespace
+
+BreedingMarket::BreedingMarket(ServerInstance& serverInstance)
+  : _serverInstance(serverInstance)
+{
+}
+
+void BreedingMarket::Initialize()
+{
+  // Get all registered stallion UIDs from the data source
+  _stallionUidsToLoad = _serverInstance.GetDataDirector().ListRegisteredStallions();
+  
+  spdlog::info("BreedingMarket::Initialize - Found {} stallion UID(s) from data source", _stallionUidsToLoad.size());
+  
+  if (_stallionUidsToLoad.empty())
+  {
+    spdlog::info("No registered stallions found in database");
+    _stallionsLoaded = true;
+    return;
+  }
+  
+  // Log the stallion UIDs
+  for (data::Uid uid : _stallionUidsToLoad)
+  {
+    spdlog::debug("  - Queuing stallion UID: {}", uid);
+  }
+  
+  spdlog::info("Queuing {} stallion(s) for async loading", _stallionUidsToLoad.size());
+  
+  // Request async loading of stallion records (individual calls trigger loading)
+  for (data::Uid stallionUid : _stallionUidsToLoad)
+  {
+    _serverInstance.GetDataDirector().GetStallionCache().Get(stallionUid);
+  }
+  
+  _stallionsLoaded = false;
+}
+
+void BreedingMarket::Terminate()
+{
+  _registeredStallions.clear();
+  _horseToStallionMap.clear();
+  _stallionDataCache.clear();
+  _stallionsLoaded = false;
+  _stallionUidsToLoad.clear();
+}
+
+void BreedingMarket::Tick()
+{
+  std::lock_guard<std::mutex> lock(_mutex);
+  
+  // Load stallions from database
+  if (!_stallionsLoaded && !_stallionUidsToLoad.empty())
+  {
+    int loadedCount = 0;
+    std::vector<data::Uid> stillPending;
+    
+    // Check each stallion individually
+    for (data::Uid stallionUid : _stallionUidsToLoad)
+    {
+      auto stallionRecordOpt = _serverInstance.GetDataDirector().GetStallionCache().Get(stallionUid);
+      if (!stallionRecordOpt)
+      {
+        // Not loaded yet, will try again next tick
+        stillPending.push_back(stallionUid);
+        continue;
+      }
+      
+      StallionData cachedData{};
+      bool isValid = true;
+      
+      uint32_t timesMated = 0;
+      
+      stallionRecordOpt->Immutable([&](const data::Stallion& stallion)
+      {
+        cachedData.stallionUid = stallion.uid();
+        cachedData.horseUid = stallion.horseUid();
+        cachedData.ownerUid = stallion.ownerUid();
+        cachedData.breedingCharge = stallion.breedingCharge();
+        cachedData.registeredAt = stallion.registeredAt();
+        timesMated = stallion.timesMated();
```

Comment:
data director's stallion cache is already your cache, why copy it to your own class?